dts/arm/st/mp2/stm32mp257.dtsi

```diff
@@ -61,6 +63,101 @@
 				      <8 1>, <9 1>, <10 1>, <11 1>,
 				      <12 1>, <13 1>, <14 1>, <15 1>;
 		};
+
+		pinctrl: pin-controller@44240000 {
+			compatible = "st,stm32-pinctrl";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0x44240000 0x2000>;
+
+			gpioa: gpio@44240000 {
+				compatible = "st,stm32-gpio";
+				gpio-controller;
+				#gpio-cells = <2>;
+				reg = <0x44240000 DT_SIZE_K(1)>;
+				clocks = <&rcc STM32_CLOCK(GPIOA, CLK)>;
```

Comment:
Because the `clocks` binding differs from other STM32, it would be required to have a dedicated MP2 RCC compatible to explain how to interpret various fields.